.github/workflows/benchmark_msbuild.yaml

```diff
@@ -0,0 +1,59 @@
+name: msbuild benchmarks
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+
+env:
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+  DOTNET_BASE_VERSION_URL_LINUX: "https://download.visualstudio.microsoft.com/download/pr/cd0d0a4d-2a6a-4d0d-b42e-dfd3b880e222/008a93f83aba6d1acf75ded3d2cfba24/dotnet-sdk-6.0.400-linux-x64.tar.gz"
+  DOTNET_BASE_VERSION_URL_WINDOWS: "https://download.visualstudio.microsoft.com/download/pr/1fb808dc-d017-4460-94f8-bf1ac83e6cd8/756b301e714755e411b84684b885a516/dotnet-sdk-7.0.100-win-x64.zip"
+  DOTNET_DAILY_VERSION_URL_LINUX: "https://aka.ms/dotnet/8.0.1xx/daily/dotnet-sdk-linux-x64.tar.gz"
+  DOTNET_DAILY_VERSION_URL_WINDOWS: "https://aka.ms/dotnet/8.0.1xx/daily/dotnet-sdk-win-x64.zip"
+  TEST_SOLUTION_REPO_URL: "https://github.com/marcin-krystianc/TestSolutions.git"
+  TEST_SOLUTION_DIR: "SimpleAppWithLibs"
+
+jobs:
+  msbuild_benchmarks:
+    strategy:
+      matrix:
+        OS: [windows-latest, ubuntu-latest]
```

Comment:
I think that the approach in `benchmarks_nuget` is a better approach (https://github.com/G-Research/DotNetPerfMonitor/blob/main/.github/workflows/benchmarks_nuget.yml#L18 and https://github.com/G-Research/DotNetPerfMonitor/blob/main/.github/workflows/benchmarks_nuget.yml#L72). We define test cases as os specific, which makes sense because some code bases which can be built on windows cannot be build on Linux and vice-versa.

Reply:
Alright, Thanks for the review again @marcin-krystianc. I was struggling to define which function to run, depending on the current os in the workflow. I will define two seperate jobs for windows and ubuntu then 🚀